src/views/sidebar/utils.ts

```diff
@@ -86,8 +87,8 @@ export const getFolderIconColorForAccordionFolder = (f: AccordionFolder): string
 	return ZIMBRA_STANDARD_COLORS[0].hex;
 };
 
-export const getFolderIconColor = (f: Folder): string => {
-	if (f?.color) {
+export const getFolderIconColor = (f: Folder | AccordionItemType): string => {
```

Comment:
unfortunately ModalAccordionCustomComponent is forced by the DS to receive an AccordionItemType. if you look at the file the getFodlerIconColor is invoked on an object that is defined as AccordionItemType but actually received as Folder. it is one of the limits to pass the custom component as a prop to the accordion, instead of as a child. so the code works but as you correctly spotted out it shouldn't.
other than the solution I came up with, to solve it on our side we could map the object, but I think it is a very expensive solution since we are talking about the entire folder structure

Reply:
Ok , I see. The type constraints set by the DS are too strict.
The solution of mapping the object is not feasible anyway because we actually need a Folder inside the custom component.
As you suggested a better solution could exist if the DS permitted to declare the accordion inner elements as children instead of props. We will discuss about it.